src/pages/organize/[orgId]/campaigns/index.tsx

```diff
@@ -33,14 +79,275 @@ export const getServerSideProps : GetServerSideProps = scaffold(async (ctx) => {
     }
 }, scaffoldOptions);
 
+const useStyles = makeStyles((theme) => ({
+    expandButton: {
+        textDecoration: 'underline',
+        textTransform: 'none',
+    },
+    speedDial: {
+        bottom: theme.spacing(4),
+        position: 'fixed',
+        right: theme.spacing(4),
+    },
+}));
+
 type AllCampaignsSummaryPageProps = {
     orgId: string;
 };
 
-const AllCampaignsSummaryPage: PageWithLayout<AllCampaignsSummaryPageProps> = () => {
+const AllCampaignsSummaryPage: PageWithLayout<AllCampaignsSummaryPageProps> = ({ orgId }) => {
+    const router = useRouter();
+    const classes = useStyles();
+    const intl = useIntl();
+    const campaignsQuery = useQuery(['campaigns', orgId], getCampaigns(orgId));
+    const upcomingEventsQuery = useQuery(['upcomingEvents', orgId], getUpcomingEvents(orgId));
+    const eventsQuery = useQuery(['events', orgId], getEvents(orgId));
+    const surveysQuery = useQuery(['surveys', orgId], getSurveys(orgId));
+    const callsQuery = useQuery(['calls', orgId], getAllCallAssignments(orgId));
+    const canvassesQuery = useQuery(['canvasses', orgId], getCanvasses(orgId));
+
+    const eventMutation = useMutation(postEvent(orgId));
+    const campaignMutation = useMutation(postCampaign(orgId));
+
+    const [expandedList, setExpandedList] = useState(false);
+    const [speedDialOpen, setSpeedDialOpen] = useState(false);
+    const [dialogOpen, setDialogOpen] = useState<null | string>(null);
+    const [filtered, setfiltered] = useState({
+        callAssignments: false,
+        canvasses: false,
+        standalones: false,
+        surveys: false,
+    });
+    const [cardHeight, setCardHeight] = useState(0);
+    const card = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        setCardHeight(card.current?.offsetHeight || 0);
+    }, []);
+
+    const campaigns = campaignsQuery.data || [];
+
+    const upcomingEvents = upcomingEventsQuery.data || [];
+    const events = eventsQuery.data || [];
+    const surveys = surveysQuery.data?.map(s => ({ ...s, key: `survey-${s.id}` })) || [];
+    const callAssignments = callsQuery.data?.map(c => ({ ...c, key: `call-${c.id}` })) || [];
+    const canvasses = canvassesQuery.data?.map(c => ({ ...c, key: `canvass-${c.id}` })) || [];
+    const standalones = events.filter(e => !e.campaign.id).map(s => ({ ...s, key: `standalone-${s.id}` }));
```

Comment:
If we put this logic in the request handlers, would the results be cached?

Reply:
I don't know, took a quick look at the docs but maybe you already know, @richardolsson ?

Reply:
Any data that the promise resolves to will be cached. But IMO it's better to cache the data that comes from the API than to cache processed data. If the processing is heavy it could be memoized in the component that processes it, but it's better for several components (with various processing needs) to be able to use the same cached raw data.

This is in many ways similar to an issue discussed earlier this year (mostly between me and Kristoffer) and that is tracked in #152.

Reply:
still don't quite understand after looking at this issue -- is it bad that we are processing this data here?
Oh, is River asking whether the logic should be moved to the fetching functions (request handlers), and if that processed data would be cached, making it better for this page -- 
But several pages might need access to this data in different forms. 
Aha, now I know what to think about when processing data inside the fetch functions!
I'm guessing based on the discussion we should leave as is

Reply:
Thanks for being able to follow along on that train of thought @nehahirve! 😊 

It sounds like you have the same understanding I do, and have come to the same conclusion. I think we should not do any processing in the fetch functions (and get rid of the processing that already exists in several such functions already as suggested by #152).

However, that doesn't mean that I think this is a great structure. To me it's a little odd to be adding these `key` attributes to the objects. The `key` is something that is specific to react, so I would expect it to be added as part of the loop that already happens in the code when the JSX is rendered.

But since this entire feature will be disabled for MVP (#217), lets leave it as is. 😊

Reply:
Great, makes sense with the key, already changed this but instead I've wrapped the objects with 'type' and data here itself, and generated the key in the JSX from the type